### PR TITLE
Update UsersModel.php for 'GROUP BY' fix

### PR DESCRIPTION
**Pull request by kochinc**
Pull Request for Issue #41951.

### Summary of Changes
Proposed fix for 
ERROR: column "mfa.mfaRecords" must appear in the GROUP BY clause or be used in an aggregate function #41951.
Please also see #41952.

### Testing Instructions
1. Log in administrator backend.
2. Select Users -> Groups on the menu.
3. Click on any entry on the "Enabled Users" or "Blocked Users" column.

### Expected result AFTER applying this Pull Request
It shows a list of enabled or blocked users with a group.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_users/src/Model/UsersModel.php [hunk 1]
```diff
@@ -385,9 +382,13 @@ protected function getListQuery()
...
-                        ]
-                    )
-                );
+                        ];
+            if (PluginHelper::isEnabled('multifactorauth')) {
+                $group_by[] = 'mfa.mfaRecords';
+            }
+
+            $query->join('LEFT', '#__user_usergroup_map AS map2 ON map2.user_id = a.id')
+                  ->group($db->quoteName($group_by));
```
Comment (richard67) [inline administrator/components/com_users/src/Model/UsersModel.php]:
```suggestion
            ];

            if (PluginHelper::isEnabled('multifactorauth')) {
                $group_by[] = 'mfa.mfaRecords';
            }

            $query->join('LEFT', '#__user_usergroup_map AS map2 ON map2.user_id = a.id')
                ->group($db->quoteName($group_by));
```
Code style: We indent 4 spaces per indent level. We do not align closing brackets for arrays with the opening bracket, we align it with the right indent level. Same for the `->`. We use an empty line before a control structure like if (in most cases).

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on fd94f2b3468c6c356cd6c15cf579bef7c163a798<br /><br />postgres<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41984">issues.joomla.org/tracker/joomla-cms/41984</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on dbc15379be659c15b206ddd86d7945e83066ac5b<br /><br />Tested with PostgreSQL 14.9 and MySQL 8.0.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41984">issues.joomla.org/tracker/joomla-cms/41984</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41984">issues.joomla.org/tracker/joomla-cms/41984</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**